### PR TITLE
Feature/cax 1 mtpdc@111 pom files

### PR DESCRIPTION
**Pull request by algattik**
Reorganized POM files to allow creating projects that are not children of the Spring Boot parent pom (e.g. the EDC connector, where Spring Boot dependency versions would conflict with Jetty etc. versions).

See https://maven.apache.org/maven-ci-friendly.html for the use of `${revision}`.

Also simplified version management.

### CODE REVIEW

**Review by ouphi — APPROVED** [review]

File: coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/repositories/PartAspectRepository.java [hunk 1]
```diff
@@ -37,5 +38,9 @@
      * @see org.springframework.data.repository.CrudRepository#findAllById(Iterable)
      */
     @Query("SELECT a FROM PartAspectEntity a WHERE a.key.partId IN (:partIds) AND a.key.name = :name")
-    List<PartAspectEntity> findAllBy(Collection<PartIdEntityPart> partIds, String name);
+    List<PartAspectEntity> findAllBy(
+            @Param("partIds")
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/repositories/PartAspectRepository.java:42]: How is this change related to the rest of this PR?

**Review by marcgs — APPROVED** [review]

File: coreservices/partsrelationshipservice/prs-parent/pom.xml [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
-        <version>2.5.4</version>
-        <relativePath/>
-        <!-- lookup parent from repository -->
+        <groupId>net.catenax.prs</groupId>
+        <artifactId>parts-relationship-service-root</artifactId>
+        <version>${revision}</version>
     </parent>
 
-    <groupId>com.catenax.prs</groupId>
+    <groupId>net.catenax.prs</groupId>
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/prs-parent/pom.xml:12]: we can also remove `groupId` declaration here as this is inherited from parent and now its redundant.

Can be done in other modules as well.

File: coreservices/partsrelationshipservice/pom.xml [hunk 3]
```diff
@@ -2,16 +2,15 @@
...
 
-    <groupId>com.catenax.prs</groupId>
+    <groupId>net.catenax.prs</groupId>
     <artifactId>parts-relationship-service-root</artifactId>
-    <version>0.0.1-SNAPSHOT</version>
+    <version>${revision}</version>
     <packaging>pom</packaging>
 
     <name>PRS Root</name>
     <description>Root module containing other modules.</description>
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/pom.xml:12]: We can declare encoding here

```
    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
    </properties>
```
Otherwise we have maven build warning:
`[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!`

File: coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/repositories/PartAspectRepository.java (hunk 1 above)
Comment (algattik) [reply coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/repositories/PartAspectRepository.java:42]: Somehow an error popped up here, perhaps one dependency ends up resolving differently. Anyway this seems to be a JPA standard requirement.

File: coreservices/partsrelationshipservice/prs-parent/pom.xml [hunk 4]
```diff
@@ -1,27 +1,28 @@
...
     <packaging>pom</packaging>
 
     <name>PRS Parent</name>
-    <description>Parent module.</description>
+    <description>Parent module for PRS modules.</description>
 
     <properties>
         <java.version>11</java.version>
+        <maven.compiler.source>${java.version}</maven.compiler.source>
+        <maven.compiler.target>${java.version}</maven.compiler.target>
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/prs-parent/pom.xml:22]:
```suggestion
        <maven.compiler.source>${java.version}</maven.compiler.source>
        <maven.compiler.target>${java.version}</maven.compiler.target>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
We can declare encoding here, Otherwise we have maven build warning:  `[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!`

**Review by cpeeyush — APPROVED** [review]
